modules/local/find_circ/find_circ/main.nf

```diff
@@ -0,0 +1,49 @@
+process FIND_CIRC {
+    tag "$meta.id"
+    label "process_high"
+
+    conda "find_circ=1.2 bowtie2"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mulled-v2-c27e472038a09e49d9147bc52903e12836302c12:60ffb3b15a2c40c669f8d38382b1e6e4b065f5e4-0' :
+        'quay.io/biocontainers/mulled-v2-c27e472038a09e49d9147bc52903e12836302c12:60ffb3b15a2c40c669f8d38382b1e6e4b065f5e4-0' }"
+
+    input:
+    tuple val(meta), path(anchors)
+    tuple val(meta2), path(index)
+    path fasta
+
+    output:
+    tuple val(meta), path("${prefix}.sites.bed"), emit: bed
+    path("${prefix}.sites.reads")               , emit: reads
+    path("${prefix}.sites.log")                 , emit: log
+    path  "versions.yml"                        , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    prefix = task.ext.prefix ?: "${meta.id}"
+    def VERSION = '1.2'
+    """
+    INDEX=`find -L ./ -name "*.rev.1.bt2" | sed "s/.rev.1.bt2//"`
+    [ -z "\$INDEX" ] && INDEX=`find -L ./ -name "*.rev.1.bt2l" | sed "s/.rev.1.bt2l//"`
+    [ -z "\$INDEX" ] && echo "Bowtie2 index files not found" 1>&2 && exit 1
+
+    bowtie2 \\
```

Comment:
even for local modules you could consider using args. It allows user later on to adjust the parameters with a custom.config if need be without having to change the pipeline code or ask you to add more pipeline parameters

Reply:
Hmm, good point. find_circ is super old, back when scripts were very eh, prescriptive